src/SFA.DAS.Reservations.Application/Reservations/Commands/CreateReservationCommandHandler.cs

```diff
@@ -16,46 +18,68 @@ namespace SFA.DAS.Reservations.Application.Reservations.Commands
 {
     public class CreateReservationCommandHandler : IRequestHandler<CreateReservationCommand, CreateReservationResult>
     {
-        private readonly IValidator<ICreateReservationCommand> _validator;
+        private readonly IValidator<IReservationQuery> _reservationQueryValidator;
+        private readonly IValidator<ICreateReservationCommand> _createReservationValidator;
         private readonly IOptions<ReservationsApiConfiguration> _apiOptions;
         private readonly IApiClient _apiClient;
         private readonly IHashingService _hashingService;
+        private readonly ICacheStorageService _cacheStorageService;
 
         public CreateReservationCommandHandler(
-            IValidator<ICreateReservationCommand> validator, 
+            IValidator<IReservationQuery> reservationQueryValidator,
+            IValidator<ICreateReservationCommand> createReservationValidator, 
             IOptions<ReservationsApiConfiguration> apiOptions,
             IApiClient apiClient,
-            IHashingService hashingService)
+            IHashingService hashingService,
+            ICacheStorageService cacheStorageService)
         {
-            _validator = validator;
+            _reservationQueryValidator = reservationQueryValidator;
+            _createReservationValidator = createReservationValidator;
             _apiOptions = apiOptions;
             _apiClient = apiClient;
             _hashingService = hashingService;
+            _cacheStorageService = cacheStorageService;
         }
 
         public async Task<CreateReservationResult> Handle(CreateReservationCommand command, CancellationToken cancellationToken)
         {
-            var validationResult = await _validator.ValidateAsync(command);
-            if (!validationResult.IsValid())
+            var queryValidationResult = await _reservationQueryValidator.ValidateAsync(command);
+            if (!queryValidationResult.IsValid())
             {
                 throw new ValidationException(
-                    new ValidationResult("The following parameters have failed validation", validationResult.ErrorList), null, null);
+                    new ValidationResult("The following parameters have failed validation", queryValidationResult.ErrorList), null, null);
             }
 
-            var startDateComponents = command.StartDate.Split("-");
+            var reservation = await _cacheStorageService.RetrieveFromCache<GetCachedReservationResult>(command.Id.ToString());
+            if (reservation == null)
+            {
+                throw new ValidationException(
```

Comment:
Will need to look at throwing a different exception at this point so we can differentiate between when the item is missing from the cache.